src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -640,5 +712,93 @@ private NamedTypeSymbol MakeAcyclicBaseType(DiagnosticBag diagnostics)
 
             return declaredBase;
         }
+
+        private TypeSymbol MakeAcyclicExtensionClassType(DiagnosticBag diagnostics)
+        {
+            if (!this.IsExtensionClass)
+            {
+                return null;
+            }
+
+            var typeKind = this.TypeKind;
+            if (typeKind != TypeKind.Class)
+            {
+                // PROTOTYPE: Temporary error message just to return some form of an ErrorTypeSymbol
+                var info = new CSDiagnosticInfo(ErrorCode.ERR_NameNotInContext, this);
+                return new ExtendedErrorTypeSymbol(this, ImmutableArray<Symbol>.Empty, LookupResultKind.Empty, info, 0);
+            }
+
+            Location extensionClassTypeLocation = null;
+            TypeSymbol extendedType = null;
+            ArrayBuilder<TypeSymbol> multipleTypesSpecified = null;
+            foreach (var decl in this.declaration.Declarations)
+            {
+                extensionClassTypeLocation = decl.Location;
+                BaseListSyntax extendedTypeList = GetBaseListOpt(decl);
+                if (extendedTypeList == null)
+                {
+                    continue;
+                }
+                var extendedTypeBinder = this.DeclaringCompilation.GetBinder(extendedTypeList);
+                foreach (var extendedTypeSyntax in extendedTypeList.Types)
+                {
+                    var typeSyntax = extendedTypeSyntax.Type;
+                    var newExtendedType = extendedTypeBinder.BindType(typeSyntax, diagnostics, null);
```

Comment:
We are now (see `ExtensionEverythingTests.Circular`). We don't actually have to do full circular analysis, since `extension` implies static, and we don't have to dig into what it extends to figure out if it's a cycle (instead we can bail upon seeing an extension class extending an extension class).